packages/toolpad-app/src/runtime/auth.tsx

```diff
@@ -1,51 +1,65 @@
 import * as React from 'react';
 import { asArray } from '@mui/toolpad-utils/collections';
-import { Box } from '@mui/material';
-
-export interface User {
-  roles?: string[];
-}
-
-export interface AuthorizationProviderProps {
-  children?: React.ReactNode;
-  user?: User | null;
-}
-
-const UserContext = React.createContext<null | User>(null);
-
-export function AuthenticationProvider({ children, user = null }: AuthorizationProviderProps) {
-  return <UserContext.Provider value={user}>{children}</UserContext.Provider>;
-}
+import { Box, CircularProgress, Container } from '@mui/material';
+import { AUTH_SIGNIN_PATH, AuthContext } from './useAuth';
 
 export interface RequireAuthorizationProps {
   children?: React.ReactNode;
   allowedRole?: string | string[];
+  basename: string;
 }
 
-export function RequireAuthorization({ children, allowedRole }: RequireAuthorizationProps) {
-  const user = React.useContext(UserContext);
+export function RequireAuthorization({
+  children,
+  allowedRole,
+  basename,
+}: RequireAuthorizationProps) {
+  const { session, isSigningIn } = React.useContext(AuthContext);
+  const user = session?.user ?? null;
+
   const allowedRolesSet = React.useMemo<Set<string>>(
     () => new Set(asArray(allowedRole ?? [])),
     [allowedRole],
   );
 
   let reason = null;
   if (!user) {
-    reason = 'User is not authenticated.';
-  } else if (!user.roles || user.roles.length <= 0) {
+    if (!isSigningIn) {
+      window.location.replace(`${basename}${AUTH_SIGNIN_PATH}`);
+    }
+
+    return (
+      <Container
+        sx={{
+          display: 'flex',
+          alignItems: 'center',
+          justifyContent: 'center',
+          flexDirection: 'column',
+          mt: 4,
+        }}
+      >
+        <CircularProgress color="primary" size={56} />
+      </Container>
+    );
+  }
+
+  if (!user.roles || user.roles.length <= 0) {
     reason = 'User has no roles defined.';
   } else if (!user.roles.some((role) => allowedRolesSet.has(role))) {
     const rolesList = user?.roles?.map((role) => JSON.stringify(role)).join(', ');
     reason = `User with role(s) ${rolesList} is not allowed access to this resource.`;
   }
 
-  return reason ? (
+  // @TODO: Once we have roles we can add back this check.
```

Comment:
What if I only want authentication, no authorization?

Reply:
I'll make sure it will still work, for that the idea is that you just need to set up an auth provider.